runtime/gc_modron_startup/mmparse.cpp

```diff
@@ -559,18 +559,152 @@ xlpSubOptionsParser(J9JavaVM *vm, IDATA xlpIndex, XlpError *xlpError, UDATA *req
+			oldPageType = getPageTypeString(requestedPageFlags);
+		}
+		if (0 == (J9PORT_VMEM_PAGE_FLAG_NOT_USED & pageFlags)) {
+			newPageType = getPageTypeString(pageFlags);
+		}
+		if (NULL == oldPageType) {
+			if (NULL == newPageType) {
+				j9nls_printf(PORTLIB, J9NLS_INFO, J9NLS_GC_OPTIONS_LARGE_PAGE_SIZE_NOT_SUPPORTED, oldSize, oldQualifier, newSize, newQualifier);
+			} else {
+				j9nls_printf(PORTLIB, J9NLS_INFO, J9NLS_GC_OPTIONS_LARGE_PAGE_SIZE_NOT_SUPPORTED_WITH_NEW_PAGETYPE, oldSize, oldQualifier, newSize, newQualifier, newPageType);
+			}
+		} else {
+			if (NULL == newPageType) {
+				j9nls_printf(PORTLIB, J9NLS_INFO, J9NLS_GC_OPTIONS_LARGE_PAGE_SIZE_NOT_SUPPORTED_WITH_REQUESTED_PAGETYPE, oldSize, oldQualifier, oldPageType, newSize, newQualifier);
+			} else {
+				j9nls_printf(PORTLIB, J9NLS_INFO, J9NLS_GC_OPTIONS_LARGE_PAGE_SIZE_NOT_SUPPORTED_WITH_PAGETYPE, oldSize, oldQualifier, oldPageType, newSize, newQualifier, newPageType);
+			}
+		}
+	}
+}
+
+/**
+ * 
+ * Obtains and sets the largest page size available on the system. 
+ * Returns true if the operation was successful.
+ * 
+ */
+bool
+gcSetPreferredPageSize(J9JavaVM *vm, MM_GCExtensions *extensions)
+{
+	UDATA defaultLargePageSize = 0;
+	UDATA defaultLargePageFlags = J9PORT_VMEM_PAGE_FLAG_NOT_USED;
+	PORT_ACCESS_FROM_JAVAVM(vm);
+
+	/* Get the default large page size of non pageable */
+	j9vmem_default_large_page_size_ex(0, &defaultLargePageSize, &defaultLargePageFlags);
+	
+	if (0 != defaultLargePageSize) {
+		extensions->requestedPageSize = defaultLargePageSize;
+		extensions->requestedPageFlags = defaultLargePageFlags;
+	} else {
+		return false;
+	}
+
+	return true;
+}
+
+/**
+ * 
+ *  Sets the large page size used within the objectheap when the options -XX:+UseLargePages or -XX:LargePageSizeInBytes=<size> are used.
+ *  The value of the large page size will only be overwritten when these arguments are located in the right most index.
+ *  This is compared to the indices of -Xlp, -Xlp<size>, and -Xlp:objectheap:pagesize=<xx>/
+ * 
+ */
+static bool
+gcProcessXXLargePageObjectHeapArguments(J9JavaVM *vm)
+{
+	PORT_ACCESS_FROM_JAVAVM(vm);
+	MM_GCExtensions *extensions = MM_GCExtensions::getExtensions(vm);
+	if (-1 == vm->largePageSizeRequested) {
+
+		UDATA defaultLargePageSize = 0;
+		UDATA defaultLargePageFlags = J9PORT_VMEM_PAGE_FLAG_NOT_USED;
+
+		/* Get the default large page size of non pageable */
+		j9vmem_default_large_page_size_ex(0, &defaultLargePageSize, &defaultLargePageFlags);
+		
+#if	defined(J9ZOS390)
+		/* Get the default large page size of pageable (ZOS Only) */
+		UDATA defaultLargePageSizePageable = 0;
+		UDATA defaultLargePageFlagsPageable = J9PORT_VMEM_PAGE_FLAG_PAGEABLE;
+
+		j9vmem_default_large_page_size_ex(0, &defaultLargePageSizePageable, &defaultLargePageFlagsPageable);
+	
+		/* Compare which of the two is larger, and use the larger default page file */
+		if (defaultLargePageSizePageable > defaultLargePageSize) {
```

Comment:
BTW there is a bug, intended to be `>=` to prefer 1m pageable compare with 1m fixed